src/core/CCSeq.ml

```diff
@@ -412,6 +416,9 @@ let to_array l =
 (*$T
   of_array [| 1; 2; 3 |] |> to_list = [1;2;3]
   of_list [1;2;3] |> to_array = [| 1; 2; 3; |]
+  let r = ref 1 in \
+  let s = unfold (fun i -> if i < 3 then let x = !r in incr r; Some (x, succ i) else None) 0 in \
+  to_array s = [| 1; 2; 3; |]
```

Comment:
I couldn't figure out a better way to do this using qtest, but happy to make any changes that would improve readability or feedback.

Reply:
that seems good.